sdks/java/extensions/jackson/src/main/java/org/apache/beam/sdk/extensions/jackson/AsJsons.java

```diff
@@ -56,6 +63,35 @@ private AsJsons(Class<? extends InputT> outputClass) {
     return newTransform;
   }
 
+  /**
+   * Returns a new {@link AsJsonsWithFailures} transform that catches exceptions raised while
+   * writing JSON elements, passing the raised exception instance and the input element being
+   * processed through the given {@code exceptionHandler} and emitting the result to a failure
+   * collection.
+   *
+   * <p>Example usage:
+   *
+   * <pre>{@code
+   * WithFailures.Result<PCollection<String>, KV<MyPojo, Map<String, String>>> result =
+   *     pojos.apply(
+   *         AsJsons.of(MyPojo.class)
+   *             .withFailures(new WithFailures.ExceptionAsMapHandler<MyPojo>() {}));
```

Comment:
Perhaps we should provide specific default exception handlers for AsJsons and ParseJsons that are identical to ExceptionAsMapHandler except that they catch only the exceptions we expect to be raised (`JsonProcessingException` for AsJsons and `IOException` for ParseJsons), reraising any other exceptions.